common/djangoapps/student/helpers.py

```diff
@@ -628,7 +629,7 @@ def do_create_account(form, custom_form=None):
         # AccountValidationError and a consistent user message returned (i.e. both should
         # return "It looks like {username} belongs to an existing account. Try again with a
         # different username.")
-        if User.objects.filter(username=user.username):
+        if username_exists_or_retired(user.username):
```

Comment:
This change is an extra one - and is arguably unneeded. If an IntegrityError happens upon a `user.save()`, it's likely because the username belongs to another user in the auth_user table itself - not in the retired user status table. However, since `user.save()` also invokes a retired user check via the pre_save signal *AND* the email check below also checks for a retired email address outside of auth_user, I made this change to keep both checks matching.